camply/providers/recreation_dot_gov/recdotgov_provider.py

```diff
@@ -625,7 +681,9 @@ def get_recdotgov_data(
             )
         return loads(response.content)
 
-    def get_campsite_by_id(self, campsite_id: int) -> CamplyModel:
+    def get_campsite_by_id(
+        self, campsite_id: int
+    ) -> Union[CampsiteResponse, TourResponse]:
```

Comment:
just `CoreRecDotGovResponse`?

Reply:
I like that idea, that was my original intent with the `CoreRecDotGovResponse` super class. I ended up going with a `Union` here instead of just using the `CoreRecDotGovResponse` return type due to some downstream MyPy Failures. (There's a lot of work that needs to be done to make this project pass MyPy tests so I'm happy with whatever)

If we want to use `CoreRecDotGovResponse` as the Return Type here we'll need to add `FacilityID` to the `CoreRecDotGovResponse` object so this snippet doesn't fail type checks: https://github.com/jingyuanliang/camply/blob/795157e18451ac681fe284ba01f9fa0614954438/camply/providers/recreation_dot_gov/recdotgov_provider.py#L722-L725